sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -128,6 +129,36 @@ public static PipelineOptions create() {
   }
 
   /**
+   * Converts a {@link PipelineOptions} to JSON that can be deserialized later using {@link
+   * #deserializeFromJson(String)}.
+   *
+   * <p>This method is intended to be used by {@link PipelineRunner runners}. Not intended for use
+   * by pipeline authors. See note "Serialization Of PipelineOptions" on {@link PipelineOptions}. Do
+   * not use this method in your transforms to circumvent the fact that {@link PipelineOptions} is
+   * not serializable or to capture the current value of {@link PipelineOptions} at pipeline
+   * construction time.
+   */
+  public static String serializeToJson(PipelineOptions options) {
+    try {
```

Comment:
<!--new_thread; commit:2e062a5fbbffbddf3d1e6a7d68656b5ab7bac3da; resolved:0-->
I'm not sure how much I like this type signature.

Perhaps this should be put in core construction? I don't think we get all that much from the use of MAPPER (feel free to correct me!)